### PR TITLE
s3 exporter: exporter improvements and migration v2 aws sdk

### PR DESCRIPTION
**Pull request by MovieStoreGuy**
#### Description

With the annoucement of [eol](https://aws.amazon.com/blogs/developer/announcing-end-of-support-for-aws-sdk-for-go-v1-on-july-31-2025/) for the aws v1 sdk, this will give the project time to adopt the latest SDK to provide feedback.

Furthermore, there is also minor code improvements to improve maintainability.

#### Testing

Included unit tests as part of the changes.

#### Documentation

This is mostly a behaviourally change, and shouldn't have any user impact but it is worth calling out.

### CODE REVIEW

**Comment by MovieStoreGuy** [comment]
Is action for https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/36699

File: exporter/awss3exporter/config.go [hunk 1]
```diff
@@ -14,16 +14,28 @@ import (
...
+	// RoleArn is the role policy to use when interacting with S3
+	RoleArn string `mapstructure:"role_arn"`
+	// S3ForcePathStyle sets the value for force path style.
+	S3ForcePathStyle bool `mapstructure:"s3_force_path_style"`
+	// DisableSLL forces communication to happen via HTTP instead of HTTPS.
+	DisableSSL bool `mapstructure:"disable_ssl"`
+	// Compression sets the algorithm used to process the payload
+	// before uploading to S3.
+	// Valid values are: `gzip` or no value set.
+	Compression configcompression.Type `mapstructure:"compression"`
```
Comment (atoulme) [inline exporter/awss3exporter/config.go:38]: unrelated to this PR, but I'll take it :D

File: exporter/awss3exporter/exporter.go [hunk 2]
```diff
@@ -7,6 +7,7 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awss3exporter/internal/upload"
```
Comment (atoulme) [inline exporter/awss3exporter/exporter.go:10]: that probably needs to move down to the next import group?

**Review by atoulme — APPROVED** [review]

**Review by atoulme — APPROVED** [review]

**Comment by rslinckx** [comment]
I recently upgraded my opentelemetry collector and ran across an issue with the new v2 aws sdk along with a google cloud storage using S3 compatibility API.

The issue is described here: https://github.com/aws/aws-sdk-go-v2/issues/1816 and also here: https://stackoverflow.com/questions/73717477/gcp-cloud-storage-golang-aws-sdk2-upload-file-with-s3-interoperability-creds

It seems the sdk v2 is no longer compatible with GCS at least without injecting some middleware code to mangle some headers.

Would you consider adding this "workaround" or is it out of scope ?

It seems there is no native GCS exporter equivalent to awss3 exporter.

**Comment by atoulme** [comment]
Moved to new issue https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/36885

**Pull request merged**